src/index.ml

```diff
@@ -112,6 +112,57 @@ module Make (K : Key) (V : Value) (IO : IO) = struct
           loop t.fans.(i) (i + 1) )
       in
       loop (-1L) 0
+
+    external set_64 : Bytes.t -> int -> int64 -> unit = "%caml_string_set64u"
+
+    external get_64 : string -> int -> int64 = "%caml_string_get64"
+
+    external swap64 : int64 -> int64 = "%bswap_int64"
+
+    let encode_int64 i =
+      let set_uint64 s off v =
+        if not Sys.big_endian then set_64 s off (swap64 v) else set_64 s off v
+      in
+      let b = Bytes.create 8 in
```

Comment:
you can allocate that buffer only once outside of the function. As you do a copy (with `Bytes.to_string`) that should be safe (unless that code is used in threads...)

Reply:
(actually this is a very bad idea in the context of multi-threaded programs :p)